test/powershell/Modules/Microsoft.PowerShell.Core/Job.Tests.ps1

```diff
@@ -22,28 +35,119 @@ Describe "Job Cmdlet Tests" -Tag "CI" {
                 $_.FullyQualifiedErrorId | should be "JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand"
             }
         }
-        It "Receive-Job can retrieve job results" -pending {
+        It "Receive-Job can retrieve job results" {
             $waitjob = Wait-Job -Timeout 10 -id $j.id
             if ( $waitJob ) {
                 $result = receive-job -id $j.id
-                $result.id | Should be 2
+                $result | Should be 2
             }
         }
     }
     Context "jobs which take time" {
+        BeforeAll {
+            $jobBlock =
+            {
+                $count = 5
+
+                while ($count)
+                {
+                    Write-Output "count: $count"
+                    --$count
+                    start-sleep 2
+                }
+            }
+        }
         BeforeEach {
-            $j = start-job -scriptblock { Start-Sleep 15 }
+            $j = start-job -scriptblock $jobBlock
         }
         AfterEach {
             Get-Job | Remove-Job -force
         }
         It "Wait-Job will wait for a job" {
             $result = wait-Job $j
             $result | should be $j
+            $j.State | should be Completed
         }
-        It "Stop-Job will stop a job" -pending {
+        It "Wait-Job will timeout waiting for a job" {
+            $result = Wait-Job -Timeout 2 $j
+            $result | should be $null
+        }
+        It "Stop-Job will stop a job" {
             Stop-Job -id $j.id
-            $j.Status |Should be Stopped
+            $j.State | should be Stopped
+        }
+        It "Remove-Job will not remove a running job" {
+            $id = $j.Id
+            $j | remove-job -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | should be $j
+        }
+        It "Remove-Job -Force will remove a running job" {
+            $id = $j.Id
+            $j | remove-job -ErrorAction SilentlyContinue
+            $job = Get-Job -Id $id
+            $job | should be $j
+            $j | Remove-Job -Force
+            $job = Get-Job -Id $id -ErrorAction SilentlyContinue
+            $job | should be $null
+        }
+    }
+    Context "Retrieving partial output from jobs" {
+        BeforeAll {
+            function getResults($job, $n, $keep)
+            {
+                $results = @()
+
+                while ($results.Count -lt $n)
```

Comment:
Here we test *-Job cmdlets so we cannot trust them.
We get `results` from Receive-Job. If Receive-Job will get the same results every call we shall get infinite loop.